lib/org/chromium/webidl/URLExtractor.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'URLExtractor',
+
+  requires: ['org.chromium.webidl.IDLFileContents'],
+
+  properties: [
+    {
+      name: 'excludeRegexp',
+      // documentation: 'Excludes all URLs matching this expression.',
+    },
+    {
+      name: 'includeRegexp',
+      // documentation: 'Only include URLs matching this expression.',
+    }
+  ],
+
+  constants: {
+    URL_REGEX: /https?:\/\/(www\.)?[-_a-zA-Z0-9.]{2,256}\.[a-z]{2,6}(\/[^?#, \n]*)?(\?[^#, \n]*)?/g,
```

Comment:
I will have to double check, but if I recall correctly, the original pattern fails a couple of tests because it matches stuff like
```
http://google.com
www.google.com
```
as one URL. I can rerun the test with the original pattern later and post the observations here.

With regards to the URL parser, Node.js has URL parsers built in and it is being used by FOAM when attempting to perform the fetch. However, we currently do not do validation on it. We can perform an additional URL parse and validation if you feel that this may be a good idea.